### ISSUE TITLE
Readiness probe fails when using custom inputs with dynamic inputs

### ISSUE THREAD

**Issue by eduardodbr** [Jeffail/benthos#751]
I'm using custom inputs that are configured using dynamic inputs. The inputs work  but then the readiness probe fails with "input not connected". I changed my custom input code to be exactly like this one: https://github.com/benthosdev/benthos-plugin-example/blob/master/input/gibberish.go and the readiness probe keeps failing.

**Comment by Jeffail** [comment]
Hey @eduardodbr, does the readiness probe fail when your config is static like this:

```yaml
input:
  dynamic:
    foo:
      type: yourthing
      plugin: ...
```

And are there any other significant aspects on the config? Like processors or the input being wrapped within another input?

I had a quick poke around and with a dynamic input the connected probe is actually always `true` https://github.com/Jeffail/benthos/blob/master/lib/broker/dynamic_fan_in.go#L123 so this is likely caused by something else in the config.

**Comment by eduardodbr** [comment]
This is my input:

```
input:
  broker:
    inputs:
      - dynamic:
          inputs: {}
          prefix: ""
          timeout: 5s
      - generate:
          mapping: root = {"url":"init"}
          count: 1
```

I thought it was caused by the dynamic input but the problem is caused by the `generate` input. While dynamic inputs are streaming data, the generate input only generates one message and closes, making the readiness probe to fail. Is this expected behavior?

**Comment by Jeffail** [comment]
Ah okay, this is an interesting one. The broker input types when asked if they're connected, which is used by the readiness probe, will return `true` if all child inputs are connected. If an input has closed then it seems reasonable to state that it's not connected, but it seems wrong that the broker would say it's not connected if has has a closed child amongst live and connected children.

We could make it so that broker inputs report they are connected when at least one child is connected. However, I think this would also be a regression as it would mean Benthos would report a successful readiness probe when we're experiencing connectivity faults in some cases.

As a proper solution we could expand the internal readiness check to distinguish between closed and not connected, and that would allow us to improve the broker response in turn. I'll mark this as an enhancement, it requires new component APIs but they can be done passively without breaking changes so this isn't blocked on V4 or anything.

**Comment by twmb** [comment]
Difficult, low demand - please open new issue / PR if still relevant & needed.
